repair error

DeathGodBXX opened the pull request:
repair expected '<document start>' but found YAML.BlockMappingStartToken at nothing error

johnnychen94 commented on demo2.jl:
```diff
@@ -0,0 +1,13 @@
+# cover: assets/sinc.png
+# ---
+
+# 代码和头信息之间必须要有空格，而且头信息前面不能有代码和注释，否则头信息不生效。
```
Could you provide a broken version that I can reproduce locally?

I don't quite understand your comment here and failed to make a broken version.

DeathGodBXX commented:
I added “# 作图” at the begining and didn’t leave empty row between “# ———” and “# 代码...”. If I deleted “# 作图”，the id，description and cover worked. If I leave empty row，no bug show.However，I didn't remember what bug I have encountered. 2469152527 邮箱：2469152527@qq.com 签名由 网易邮箱大师 定制 On 08/29/2020 23:20, Johnny Chen wrote: @johnnychen94 commented on this pull request. In demo2.jl: > @@ -0,0 +1,13 @@

+# ---

+# id: demo2

+# description: plot sinc(0--2pi) picture

+# cover: assets/sinc.png

+# ---

+

+# 代码和头信息之间必须要有空格，而且头信息前面不能有代码和注释，否则头信息不生效。

 Could you provide a broken version that I can reproduce locally? I don't quite understand your comment here and failed to make a broken version. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

DeathGodBXX commented:
If you didn't reproduce that broken version again，please tell me.Tomorrow，I run my code again and mail you。 2469152527 邮箱：2469152527@qq.com 签名由 网易邮箱大师 定制 On 08/30/2020 00:13, 2469152527@qq.com wrote: I added “# 作图” at the begining and didn’t leave empty row between “# ———” and “# 代码...”. If I deleted “# 作图”，the id，description and cover worked. If I leave empty row，no bug show.However，I didn't remember what bug I have encountered. 2469152527 邮箱：2469152527@qq.com 签名由 网易邮箱大师 定制 On 08/29/2020 23:20, Johnny Chen wrote: @johnnychen94 commented on this pull request. In demo2.jl: > @@ -0,0 +1,13 @@

+# ---

+# id: demo2

+# description: plot sinc(0--2pi) picture

+# cover: assets/sinc.png

+# ---

+

+# 代码和头信息之间必须要有空格，而且头信息前面不能有代码和注释，否则头信息不生效。

 Could you provide a broken version that I can reproduce locally? I don't quite understand your comment here and failed to make a broken version. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

johnnychen94 commented:
Thanks for the exaplaination, I can now confirm that

```
# /usr/bin/julia
# ---
# title: plot example
# ---
```

fails to parse the YAML frontmatter. This is a bit tricky to fix because it is very hard to check whether this `# ---` should be recognized as frontmatter; frontmatter is expected to be placed at the first line. I'll see what I can do to improve the status quo tomorrow.

>  didn’t leave empty row between “# ———” and “# 代码...”.

I still can't reproduce this, I tried

```
# ---
# title: plot example
# ---
# some codes
```

and

```
# ---
# title: plot example
# ---
some codes
```

and it's parsed normally. It would be great if you can update this PR with a broken example.

Thanks again for the feedback.

johnnychen94 commented:
closed together with #43 

Thanks for the bug report.
